### PR TITLE
Added identity folder and ClaimsIdentityRpc.proto.

### PR DESCRIPTION
**Pull request by kashimiz**
Adds definitions for ClaimsPrincipal binding on RpcHttp. New message definitions are separated into their own subfolder which can be used for other additional definitions going forward.

### CODE REVIEW

File: src/proto/FunctionRpc.proto [hunk 1]
```diff
@@ -10,6 +10,7 @@ option go_package ="github.com/Azure/azure-functions-go-worker/internal/rpc";
 package AzureFunctionsRpcMessages;
 
 import "google/protobuf/duration.proto";
+import "deps/IdentityRpc.proto";
```
Comment (pragnagopa) [inline src/proto/FunctionRpc.proto]: Following the conventions similar to https://github.com/googleapis/googleapis/tree/master/google/bytestream, move new definition to its own folder

File: src/proto/deps/IdentityRpc.proto [hunk 2]
```diff
@@ -0,0 +1,20 @@
+syntax = "proto3";
+// protobuf vscode extension: https://marketplace.visualstudio.com/items?itemName=zxh404.vscode-proto3
+
+// Light-weight representation of a .NET System.Security.Claims.ClaimsIdentity object.
+// This is the same serialization as found in EasyAuth, and needs to be kept in sync with
```
Comment (pragnagopa) [inline src/proto/deps/IdentityRpc.proto]: Nit: Add link to EasyAuth  and ClaimsIdentitySlim doc

File: src/proto/deps/IdentityRpc.proto [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+// its ClaimsIdentitySlim definition.
+message RpcIdentity {
```
Comment (pragnagopa) [inline src/proto/deps/IdentityRpc.proto]: Nit: RpcClaimsIdentity

File: src/proto/deps/IdentityRpc.proto [hunk 4]
```diff
@@ -0,0 +1,20 @@
+syntax = "proto3";
+// protobuf vscode extension: https://marketplace.visualstudio.com/items?itemName=zxh404.vscode-proto3
```
Comment (pragnagopa) [inline src/proto/deps/IdentityRpc.proto]: - Rename file to ClaimsIdentityRpc.proto to be more descriptive
- Move to folder: Identity

File: src/proto/FunctionRpc.proto (hunk 1 above)
Comment (kashimiz) [reply src/proto/FunctionRpc.proto]: Moved to `identity` folder.

File: src/proto/deps/IdentityRpc.proto (hunk 2 above)
Comment (kashimiz) [reply src/proto/deps/IdentityRpc.proto]: Added link to ClaimsIdentitySlim definition in Azure/azure-webjobs-sdk-extensions. I don't believe the EasyAuth definition is publicly available.

**Pull request merged**